internal/envoy/version_test.go

```diff
@@ -154,3 +213,10 @@ func TestCurrentVersion_Validates(t *testing.T) {
 		require.EqualError(t, err, fmt.Sprintf(`invalid version in "$ENVOY_VERSION": "c.c.c" should look like "%s"`, version.LastKnownEnvoy))
 	})
 }
+
+func requireWorkingDirVersion(t *testing.T, v string) (removeTempDir func()) {
+	tempDir, removeTempDir := morerequire.RequireNewTempDir(t)
```

Comment:
I'm a bit surprised to see that removeTempDir is properly assigned even though it looks like `removeTempDir` is shadowed here by `:=`.. I didn't know this behavior..https://play.golang.org/p/m0H6KZAm6aw

Reply:
I'll try make it more explicit.. this is a bit of a pain because we can leak a temp dir if not careful..

Reply:
ps this is another reason why I had it sequentially before :D anyway this function will look terrible but more clear

Reply:
I extracted a new function for this